### PR TITLE
Adding additional multicall usage

### PR DESCRIPTION
**Pull request by IanWoodard**
Given the success seen in the previous PR #310, I decided to move forward and add multicall to more places across the codebase. This change, in particular, helps us drastically cut down requests on the portfolio and markets page, causing a roughly 50% decrease in the number of outgoing requests.

### CODE REVIEW

File: earn/src/data/LendingPair.ts [hunk 1]
```diff
@@ -77,79 +80,153 @@ export async function getAvailableLendingPairs(
...
+  const contractCallContexts: ContractCallContext[] = [];
+
+  addresses.forEach((market) => {
+    contractCallContexts.push({
+      reference: `basics0-${market.pool}`,
+      contractAddress: ALOE_II_KITTY_LENS_ADDRESS,
+      abi: KittyLensABI,
+      calls: [
+        {
+          reference: `basics0-${market.pool}`,
```
Comment (haydenshively) [inline earn/src/data/LendingPair.ts:92]: I don't think individual calls should have the same `reference` id as the outer contract. I guess if it works it works though.

File: earn/src/data/LendingPair.ts [hunk 2]
```diff
@@ -77,79 +80,153 @@ export async function getAvailableLendingPairs(
...
-      const APY1 = utilization1 * (1 - 1 / 8) * (interestRate1.div(10 ** 12).toNumber() ** (365 * 24 * 60 * 60) - 1.0);
-
-      let IV = oracleResult[1].div(1e9).toNumber() / 1e9;
-      // Annualize it
-      IV *= Math.sqrt(365);
-
-      return new LendingPair(
+  const contractCallContexts: ContractCallContext[] = [];
+
+  addresses.forEach((market) => {
```
Comment (haydenshively) [inline earn/src/data/LendingPair.ts:85]: At some point this has got to break right? I can't imagine we can stuff 100 different lender requests into a single multicall, for example. Maybe worthwhile to paginate this somehow?

File: earn/src/data/LendingPair.ts (hunk 1 above)
Comment (IanWoodard) [reply earn/src/data/LendingPair.ts:92]: I honestly threw this together somewhat hastily, so I will think about some other ways to do this.

File: earn/src/data/LendingPair.ts (hunk 2 above)
Comment (IanWoodard) [reply earn/src/data/LendingPair.ts:85]: I just load-tested it and was able to do 6750 total calls without it failing. With that said, each lender does require more than one call, but that should still give us plenty of breathing room.

**Review by haydenshively — DISMISSED** [review]

**Comment by IanWoodard** [comment]
Revamped the logic to make it more precise and robust. Working towards the ideal way of doing multi-calls.

**Review by haydenshively — APPROVED** [review]

**Pull request merged**